File: app/src/main/java/com/limelight/preferences/PerfOverlayOrientationPreference.java
package com.limelight.preferences;

import android.content.Context;
import android.preference.ListPreference;
import android.preference.PreferenceManager;
import android.util.AttributeSet;

public class PerfOverlayOrientationPreference extends ListPreference {
    
    public PerfOverlayOrientationPreference(Context context, AttributeSet attrs, int defStyleAttr, int defStyleRes) {
        super(context, attrs, defStyleAttr, defStyleRes);
    }

    public PerfOverlayOrientationPreference(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    public PerfOverlayOrientationPreference(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public PerfOverlayOrientationPreference(Context context) {
        super(context);
    }

    @Override
    protected void onDialogClosed(boolean positiveResult) {
        super.onDialogClosed(positiveResult);
        
        if (positiveResult) {
            // 当方向改变时，通知位置Preference更新选项
            updatePositionPreference();
        }
    }
    
    private void updatePositionPreference() {
        PreferenceManager preferenceManager = getPreferenceManager();
        if (preferenceManager != null) {
            DynamicPerfOverlayPositionPreference positionPref = 
                (DynamicPerfOverlayPositionPreference) preferenceManager.findPreference("list_perf_overlay_position");
            if (positionPref != null) {
                positionPref.refreshEntries();
            }
        }
    }
    

} 

File: app/src/main/java/com/limelight/preferences/PerfOverlayOrientationPreference.kt
package com.limelight.preferences

import android.content.Context
import androidx.preference.ListPreference
import androidx.preference.Preference
import android.util.AttributeSet

class PerfOverlayOrientationPreference : ListPreference {
    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int, defStyleRes: Int) :
        super(context, attrs, defStyleAttr, defStyleRes) { init() }

    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) :
        super(context, attrs, defStyleAttr) { init() }

    constructor(context: Context, attrs: AttributeSet?) :
        super(context, attrs) { init() }

    constructor(context: Context) :
        super(context) { init() }

    private fun init() {
        onPreferenceChangeListener = Preference.OnPreferenceChangeListener { _, _ ->
            updatePositionPreference()
            true
        }
    }

    private fun updatePositionPreference() {
        val preferenceManager = preferenceManager ?: return
        val positionPref = preferenceManager.findPreference("list_perf_overlay_position")
                as? DynamicPerfOverlayPositionPreference
        positionPref?.refreshEntries()
    }
}
